DashAI/front/src/components/experiments/PrepareDatasetStep.jsx

```diff
@@ -0,0 +1,398 @@
+    ) {
+      setSplitsReady(true);
+    } else if (
+      rowsPreference === "random" &&
+      !rowsPartitionsError &&
+      rowsPartitionsPercentage.training > 0 &&
+      rowsPartitionsPercentage.validation > 0 &&
+      rowsPartitionsPercentage.testing > 0
+    ) {
+      setSplitsReady(true);
+    } else {
+      setSplitsReady(false);
+    }
+  }, [rowsPartitionsIndex, rowsPartitionsPercentage, rowsPartitionsError]);
+  useEffect(() => {
+    if (columnsReady && splitsReady) {
+      setNewExp({
+        ...newExp,
+        input_columns: inputColumns,
+        output_columns: outputColumns,
+        splits:
+          rowsPreference === "manually"
+            ? rowsPartitionsIndex
+            : rowsPartitionsPercentage,
+      }); // splits should depend on preference
+      setNextEnabled(true);
+    } else {
+      setNextEnabled(false);
+    }
+  }, [columnsReady, splitsReady]);
+
+  return (
+    <React.Fragment>
+      <Grid container spacing={1}>
+        <Grid item xs={12}>
+          <Typography item variant="subtitle1" component="h3" sx={{ mb: 0 }}>
+            Choose which columns to use
+          </Typography>
+        </Grid>
+        <Grid item xs={12}>
+          <Typography
+            item
+            variant="caption"
+            component="h3"
+            sx={{ mb: 2, color: "grey" }}
+          >
+            Ex: 1-6, 23-108
+          </Typography>
+        </Grid>
+
+        <TextField
+          required
+          id="dataset-input-columns"
+          label="Input"
+          fullWidth
+          autoComplete="off"
+          onChange={handleInputColumnsChange}
+          error={parseInputColumnsError}
+          helperText={parseInputColumnsError ? parseInputColumnsErrorText : ""}
+          sx={{ mb: 2 }}
+        />
+        <TextField
+          required
+          id="dataset-output-columns"
+          label="Output"
+          fullWidth
+          autoComplete="off"
+          onChange={handleOutputColumnsChange}
+          error={parseOutputColumnsError}
+          helperText={
+            parseOutputColumnsError ? parseOutputColumnsErrorText : ""
+          }
+          sx={{ mb: 2 }}
+        />
+        <Grid container spacing={1}>
+          <Grid item xs={12}>
+            <Typography variant="subtitle1" component="h3" sx={{ mb: 2 }}>
+              Choose how you wanna divide the dataset for the experiment
+            </Typography>
+          </Grid>
+        </Grid>
+        <RadioGroup
+          aria-labelledby="demo-radio-buttons-group-label"
+          defaultValue={"random"}
+          name="radio-buttons-group"
+          onChange={handleRowsPreferenceChange}
+        >
+          <FormControlLabel
+            value="random"
+            control={<Radio />}
+            label="Use random rows by percentage"
+          />
+          {rowsPreference === "random" ? (
+            <React.Fragment>
+              <Grid container direction="row" spacing={4}>
```

Comment:
The options look very close together. How about adding some small margins?

```suggestion
              <Grid container direction="row" spacing={4} sx={{my: 2}}>
```